### ISSUE TITLE
`PlatformTracing` behaviour of `platform_key` on interface field

### ISSUE THREAD

**Issue by ravangen** [rmosolgo/graphql-ruby#4292]
**Describe the bug**

Due to platform tracing caching, when an interface's field is being traced, the first usage of a field will use it's parent type's graphql name, instead of the interface's. When a request has different objects that implement the same interface, you can have the `platform_key` incorrectly reported.

https://github.com/rmosolgo/graphql-ruby/blob/ad8c2e1f0f71cfec6e157ad006015c552daec73e/lib/graphql/tracing/platform_tracing.rb#L43

https://github.com/rmosolgo/graphql-ruby/blob/ad8c2e1f0f71cfec6e157ad006015c552daec73e/lib/graphql/tracing/statsd_tracing.rb#L29-L31

**GraphQL schema**

```ruby
module Node
  include GraphQL::Schema::Interface
  field :id, String, null: false
end

class Image < GraphQL::Schema::Object
  implements Node
end

class Product < GraphQL::Schema::Object
  implements Node
end

class QueryRoot < GraphQL::Schema::Object
  field :nodes, [Node], null: false

  def nodes
    [
      { id: "gid://example/Image/1" },
      { id: "gid://example/Product/1" },
      { id: "gid://example/Product/2" },
    ]
  end
end

class ApplicationSchema < GraphQL::Schema
  query QueryRoot
  use GraphQL::Tracing::StatsdTracing
  # …
end
```

**GraphQL query**

Example GraphQL query and response (if query execution is involved)

```graphql
query {
  nodes { id }
}
```

```json
{
  "data": {
    "nodes": […]
  }
}
```

**Expected behavior**

```json
"graphql.Image.id"
"graphql.Product.id"
"graphql.Product.id"
```

OR

```json
"graphql.Node.id"
"graphql.Node.id"
"graphql.Node.id"
```

**Actual behavior**

```json
"graphql.Image.id"
"graphql.Image.id"
"graphql.Image.id"
```

**Additional context**

This is a minor bug / unexpected behaviour. I am not necessarily saying it needs to change, but might be at least worth a warning in the docs.

The cache key is only based on `field`, but the value is based on both `data[:owner]` and `field`, which makes me think we could use `field.owner` instead (resulting in `Node` instead of `Image`/`Product`/...)?

```diff
- "graphql.#{type.graphql_name}.#{field.graphql_name}" 
+ "graphql.#{field.owner.graphql_name}.#{field.graphql_name}" 
```

**Comment by rmosolgo** [comment]
Thanks for the detailed report, and sorry for the trouble!

>  use `field.owner` instead 

Yes, I think you're right about this. Old GraphQL versions didn't have `field.owner` which is probably why it isn't used here.

**Comment by rmosolgo** [comment]
I'm working up a better-performing tracing API over in #4344 and I'll make sure this is addressed there! (107fcc232)

**Comment by rmosolgo** [comment]
:+1: Updated platform_tracing to use `field.owner` in a54d13128, this gives `Node.id` in the trace, since `field :id` belongs to `module Node ...`.

**Comment by rmosolgo** [comment]
Fixed by #4344
